### ISSUE TITLE
Activation Function 選擇

### ISSUE THREAD

**Issue by bobolee1239** [2020-DL-Training-Program/Lab1-Backpropagation#1]
助教您好，

我想請問這次作業對 activation function 選擇上有限制一定要 sigmoid 嗎？
還是可以選擇用 ReLU 呢？

Thanks!

**Comment by zivzone** [comment]
@bobolee1239 你好，按照 Lab1 的 [spec pdf](https://github.com/2020-DL-Training-Program/Lab1-Backpropagation/blob/master/Lab1-Backpropagation_v5.pdf) "Implementation detail" 中有提到 activation function 為 sigmoid，因此還請使用 sigmoid，但也可以在報告中多試其他的並進行比較

**Comment by bobolee1239** [comment]
好的。
另外，我有發現如果用 sigmoid 當作 activation function，跟 linear case 的比起來，XOR 的 case 好像不是很好 train，請問助教這是有什麼原因嗎？(我猜想大概是微分值太小不好練，因為 ReLU 沒有這個問題 XD)

**Comment by zivzone** [comment]
@bobolee1239 你好，應該不會 train 不起來，但 train 不起來可能有很多原因。先假設一切實做沒有問題，那或許可以試試調大 learning rate 等等

**Comment by zivzone** [comment]
@bobolee1239 但你的 case 是 relu 可以 train 起來但 sigmoid 不行嗎? 如果是的話，可以我們可能也需要看看你的 code 實作上是不是有甚麼問題，下禮拜三的 lab 課可以帶來找助教詢問一下

**Comment by bobolee1239** [comment]
Hi 助教你好，我有train起來，但相對linear case 比較難 train 想問問看可能的原因這樣，因爲用 relu 好像兩個case很好練 XD

**Comment by zivzone** [comment]
@bobolee1239 喔喔那就好XD 印象中 XOR 的 case 應該是比 linear 的還難 train 沒錯 (至少參數等等要調整一下)但應該都可已很快 train 完。 report 裡可以提到這件事並且分析一下。
